tests/neural_compressor/test_optimization.py

```diff
@@ -278,6 +307,41 @@ def test_dynamic_diffusion_model(self):
         # Compare model outputs
         self.assertTrue(np.allclose(loaded_pipe_outputs, outputs, atol=1e-4))
 
+    def test_quantize_text_generate_model_quantize(self):
+        model_id = "EleutherAI/gpt-neo-125m"
+        set_seed(42)
+        model = AutoModelForCausalLM.from_pretrained(model_id)
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        tokens = tokenizer("This is a sample", return_tensors="pt")
+        input_ids = model.dummy_inputs["input_ids"]
+        input_bs, input_len = input_ids.shape
+        past_key_values = generate_dummy_past_key_values(input_bs, model)
+        attention_mask = torch.ones(input_bs, input_len + 1)
+        attention_mask[:, 0] = 0
+        example_inputs = (
+            input_ids,
+            tuple(past_key_values),
+            attention_mask,
+        )
+
+        def calibration_fn(p_model):
+            tmp_model = INCModelForCausalLM(p_model, model.config)
+            tmp_model.generate(**tokens, max_new_tokens=32, do_sample=False)
+
+        quantization_config = PostTrainingQuantConfig(approach="static", example_inputs=example_inputs)
+        model.config.return_dict = False
+        quantizer = INCQuantizer.from_pretrained(model, calibration_fn=calibration_fn)
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            quantizer.quantize(
+                quantization_config=quantization_config,
+                save_directory=tmp_dir,
+                save_onnx_model=False,
+            )
+            model = INCModelForCausalLM.from_pretrained(tmp_dir, export=True)
+
+        outputs = model.generate(**tokens, do_sample=False, num_beams=1, temperature=0.9, min_length=20, max_length=20)
+        self.assertIsInstance(outputs, torch.Tensor)
```

Comment:
It wuld be nice to compare the output of the quantizer model with the loaded `INCModelForCausalLM` model

Reply:
Done